airflow/contrib/example_dags/example_qubole_operator.py

```diff
@@ -198,7 +198,7 @@ def compare_result(ds, **kwargs):
 
 /** Computes an approximation to pi */
 object SparkPi {
-  def main(args: Array[String]) {
+  def main(args) {
```

Comment:
This one is a false positive -- this is Scala code :D

Reply:
Aaaaah indeed. So I need to add a a possibility to exclude such false positives. Will do soon.

Reply:
Looks like https://pre-commit.com/#regular-expressions to exclude the whole file is the only way to do this one. A bit overly broad but this file isn't changed very often so that's probably okay. That or adjust the regex to not allow `) {` on the line (trickier regex to get that working)

Reply:
Actually I thought about writing a bit more complex regex to exclude lines which have the standard `# noqa` in the same line - so that you can explicitly disable it if needed.

Here is the regex that will likely work :)

`^(?!.*# noqa)(.*\.super\(\)|^\s*def\s*\S*\([^):]*:.*|^\sdef\s*\S*\(.*\):\s*\-\>\s*\S*).*$
`

Reply:
And likely with (?x) I can write in multiple lines to make it easier to read :)

Reply:
But I also remember about "you have one problem - introduce regexp - then you will have two problems" saying, so maybe it's even worth to write our own simple hook for that purpose if you think it's too "nerdy" with the regexp.

Reply:
I used to work as a Perl developer for about 5 years, so I am in favour of regeps :D You can take the programmer out of perl, but not the perl out of the programmer.

Reply:
If each pygrep test is quick, might it be easier to split the super case from the type annotation case into separate rules?

Reply:
Actually if you are a perl programmer - then you will find the final result well.. refreshing :).

I found that with (?x) extension you can not only write elegant and pretty "modular" code but also one that can contain comments (!).

Also the exclusion rules can be rather elegant :).

Surprisingly I have now regexp that I am actually proud of!